File: week05/snippets/opa_properties.mjs
import fs from 'node:fs';
import BigJSON from 'big-json';

// Load the data from the GeoJSON file
const data = await BigJSON.parse({
  body: fs.readFileSync('data/opa_properties_public.geojson')
});

// Write the data to a JSONL file
const f = fs.createWriteStream('data/opa_properties_public.jsonl');
for (const feature of data.features) {
  const row = feature.properties;
  if (feature.geometry && feature.geometry.coordinates) {
    row.geog = JSON.stringify(feature.geometry);
  } else {
    row.geog = null;
  }
  f.write(JSON.stringify(row) + '\n');
}

File: week05/snippets/opa_properties.py
import json


# Load the data from the GeoJSON file
with open('data/opa_properties_public.geojson', 'r') as f:
    data = json.load(f)


# Write the data to a JSONL file
with open('data/opa_properties_public.jsonl', 'w') as f:
    for feature in data['features']:
        row = feature['properties']
        if feature['geometry'] and feature['geometry']['coordinates']:
            row['geog'] = json.dumps(feature['geometry'])
        else:
            row['geog'] = None
        f.write(json.dumps(row) + '\n')
